src/js/components/AppListFilterComponent.jsx

```diff
@@ -59,13 +55,25 @@ var AppListFilterComponent = React.createClass({
         filterText: filterText,
         activated: filterText !== "" || state.focused
       });
-      this.props.onChange({[FilterTypes.TEXT]: filterText});
+    }
+  },
+
+  updateFilters: function (key, value) {
+    var router = this.context.router;
+    var params = router.getCurrentParams();
+    var query = router.getCurrentQuery();
+
+    if (key != null && value != null) {
+      query[key] = value;
+    }
+
+    if (params != null && query.filterText !== "") {
```

Comment:
The `query.filterText !== ""` prevents the ESC-key to clear this filter.

One solution could be this:
```javascript
  updateFilters: function (key, value) {
    var router = this.context.router;
    var params = router.getCurrentParams();
    var query = router.getCurrentQuery();

    if (key != null && value != null) {
      if (value === "") {
        delete query[key];
      } else {
        query[key] = value;
      }
    }

    if (params != null) {
      router.transitionTo("apps", params, query);
    }
  },
```